src/Series.cpp

```diff
@@ -1022,13 +1022,13 @@ SeriesInterface::readGorVBased( bool do_init )
 
     auto readSingleIteration =
         [&series, &pOpen, this]
-        (uint64_t index, std::string path, bool guardClosed )
+        (uint64_t index, std::string path, bool guardAgainstRereading )
     {
         if( series.iterations.contains( index ) )
         {
             // maybe re-read
             auto & i = series.iterations.at( index );
-            if( guardClosed && i.closedByWriter() )
+            if( guardAgainstRereading && i.written() )
```

Comment:
I think that this logic (already before this PR) breaks deferred parsing in the following situation:
* group-based interation encoding
* no steps enabled in the backend
* streaming-based API used for reading

I have a feeling that `Series::readGorVBased` will read all iterations upon the second time being called.

Reply:
No, all is fine. I've added a test for it.

Reply:
@franzpoeschel comment for the bug